subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -104,3 +512,259 @@ u8_t ll_adv_aux_set_clear(void)
+{
+	return aux_handle_get((void *)lll->hdr.parent);
+}
+
+u32_t ull_adv_aux_start(struct ll_adv_aux_set *aux, u32_t ticks_anchor,
+			u32_t volatile *ret_cb)
+{
+	u32_t slot_us = EVENT_OVERHEAD_START_US + EVENT_OVERHEAD_END_US;
+	u32_t ticks_slot_overhead;
+	u8_t aux_handle;
+	u32_t ret;
+
+	/* TODO: Calc AUX_ADV_IND slot_us */
+	slot_us += 1000;
+
+	/* TODO: active_to_start feature port */
+	aux->evt.ticks_active_to_start = 0;
+	aux->evt.ticks_xtal_to_start =
+		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US);
+	aux->evt.ticks_preempt_to_start =
+		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_PREEMPT_MIN_US);
+	aux->evt.ticks_slot = HAL_TICKER_US_TO_TICKS(slot_us);
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_LOW_LAT)) {
+		ticks_slot_overhead = MAX(aux->evt.ticks_active_to_start,
+					  aux->evt.ticks_xtal_to_start);
+	} else {
+		ticks_slot_overhead = 0;
+	}
+
+	aux_handle = aux_handle_get(aux);
+
+	*ret_cb = TICKER_STATUS_BUSY;
+	ret = ticker_start(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
+			   (TICKER_ID_ADV_AUX_BASE + aux_handle),
+			   ticks_anchor, 0,
+			   HAL_TICKER_US_TO_TICKS((u64_t)aux->interval * 625),
+			   TICKER_NULL_REMAINDER, TICKER_NULL_LAZY,
+			   (aux->evt.ticks_slot + ticks_slot_overhead),
+			   ticker_cb, aux,
+			   ull_ticker_status_give, (void *)ret_cb);
+
+	return ret;
+}
+
+u8_t ull_adv_aux_stop(struct ll_adv_aux_set *aux)
+{
+	volatile u32_t ret_cb = TICKER_STATUS_BUSY;
+	u8_t aux_handle;
+	void *mark;
+	u32_t ret;
+
+	mark = ull_disable_mark(aux);
+	LL_ASSERT(mark == aux);
+
+	aux_handle = aux_handle_get(aux);
+
+	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
+			  TICKER_ID_ADV_AUX_BASE + aux_handle,
+			  ull_ticker_status_give, (void *)&ret_cb);
+
+	ret = ull_ticker_status_take(ret, &ret_cb);
+	if (ret) {
+		mark = ull_disable_mark(aux);
+		LL_ASSERT(mark == aux);
+
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	ret = ull_disable(&aux->lll);
+	LL_ASSERT(!ret);
+
+	mark = ull_disable_unmark(aux);
+	LL_ASSERT(mark == aux);
+
+	aux->is_started = 0U;
+
+	return 0;
+}
+
+void ull_adv_aux_offset_get(struct ll_adv_set *adv)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, mfy_aux_offset_get};
+	u32_t ret;
+
+	mfy.param = adv;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH, TICKER_USER_ID_ULL_LOW, 1,
+			     &mfy);
+	LL_ASSERT(!ret);
+}
+
+static int init_reset(void)
+{
+	/* Initialize adv aux pool. */
+	mem_init(ll_adv_aux_pool, sizeof(struct ll_adv_aux_set),
+		 sizeof(ll_adv_aux_pool) / sizeof(struct ll_adv_aux_set),
+		 &adv_aux_free);
+
+	return 0;
+}
+
+static inline struct ll_adv_aux_set *aux_acquire(void)
+{
+	return mem_acquire(&adv_aux_free);
+}
+
+static inline void aux_release(struct ll_adv_aux_set *aux)
+{
+	mem_release(aux, &adv_aux_free);
+}
+
+static inline u16_t aux_handle_get(struct ll_adv_aux_set *aux)
+{
+	return mem_index_get(aux, ll_adv_aux_pool,
+			     sizeof(struct ll_adv_aux_set));
+}
+
+static void mfy_aux_offset_get(void *param)
```

Comment:
As far as I understand, this function determines where to place the auxiliary advertisement. It seems to be quite elaborate for what it does. Can you maybe explain why this is the preferred way of doing this?

Reply:
no action.